homeassistant/components/hypontech/config_flow.py

```diff
@@ -0,0 +1,132 @@
+"""Config flow for the Hypontech Cloud integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from hyponcloud import AuthenticationError, HyponCloud
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    session = async_get_clientsession(hass)
+    hypon = HyponCloud(data[CONF_USERNAME], data[CONF_PASSWORD], session)
+    try:
+        if not await hypon.connect():
+            raise CannotConnect
+    except TimeoutError:
+        raise CannotConnect from None
+    except AuthenticationError:
+        raise InvalidAuth from None
+
+
+class HypontechConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hypontech Cloud."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # Use username as unique_id to prevent duplicate config entries
+                await self.async_set_unique_id(
+                    user_input[CONF_USERNAME].strip().lower()
+                )
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title="Hypontech micro inverter",
+                    data=user_input,
+                )
```

Comment:
I think the title needs to be different, if I have multiple accounts for whatever reason, I will just have a list of `Hypontech micro inverter` config entries which isn't immediately clear to the end user what entry is related to what account.

Reply:
I've checked a few other integration in core and all have default title. Title can be easily changed and it makes sense for a concise title for 99.9% of users. They are more likely to have multiple devices in the same account (but here we only support the virtual "overview" device).

Reply:
For example APSystems (another micro inverter brand) uses just [Solar](https://github.com/home-assistant/core/blob/dev/homeassistant/components/apsystems/config_flow.py#L50) as name, or Airthings just uses [Airthings](https://github.com/home-assistant/core/blob/dev/homeassistant/components/airthings/config_flow.py#L65).

You are correct that many use a specific title. I'm wondering what would be a better choice here, since a specific title complicates (as sensor names are prefixed by title) the most users' life.

Reply:
I see, actually the entity names (and IDs) should be prefixed by the device name, not the config entry title if configured correctly.

For example:

<img width="1062" height="335" alt="image" src="https://github.com/user-attachments/assets/1bf61c0b-cd07-4b6e-b7f2-aaea2ed7f75b" />

And then in the device

<img width="587" height="502" alt="image" src="https://github.com/user-attachments/assets/20c9258e-9bd3-4aa5-86a7-ea9527681b44" />

Reply:
Thank you. Indeed in the code it uses the config title as device title. I've fixed that: use specific config title, and use a shorter device title (just "Hypontech" as this is the unique device for now).